File: programmers/programmers04.py
from collections import Counter
def solution(k, tangerine):
    answer = 0
    dict_tan = dict(Counter(tangerine))
    dict_tan = sorted(dict_tan.items(), key = lambda x : x[1], reverse = True)
    
    for cnt in dict_tan:
        if k - cnt[1] <= 0:
            return answer + 1
        else:
            k -= cnt[1]
            answer += 1
    return answer

File: java/programmers04.java
class Solution {
    public int solution(String A, String B) {
        int answer = 0;
        if (A.equals(B)){
            return 0;
        }
        else{
            int len = A.length();
            for (int i = 0; i < len; i++){
                B = B.substring(1, len) + B.substring(0, 1);
                if (A.equals(B)){
                    answer = i + 1;
                    return answer;
                }
            }
            return -1;
        }
    }
}